test/byContract/IexecPoco/01_initialize.test.ts

```diff
@@ -0,0 +1,214 @@
+                volume: 9,
+            };
+            // Request order is matched in 2 deals
+            const { dealId: dealId0 } = await iexecWrapper.signAndMatchOrders({
+                app: appOrder,
+                dataset: datasetOrder,
+                workerpool: workerpoolOrder0,
+                requester: requestOrder,
+            });
+            const { dealId: dealId1 } = await iexecWrapper.signAndMatchOrders({
+                app: appOrder,
+                dataset: datasetOrder,
+                workerpool: workerpoolOrder1,
+                requester: requestOrder,
+            });
+            expect((await iexecPoco.viewDeal(dealId0)).botFirst).equal(0);
+            expect((await iexecPoco.viewDeal(dealId1)).botFirst).equal(1);
+
+            // Will fail since passed taskIndex is below deal1.botFirst
+            await expect(iexecPocoAsAnyone.initialize(dealId1, 0)).to.be.reverted;
+        });
+
+        it('Should not initialize since index is too high', async function () {
+            const { orders } = buildOrders({
+                assets: ordersAssets,
+                requester: requester.address,
+                prices: ordersPrices,
+            });
+            const { dealId } = await iexecWrapper.signAndMatchOrders(orders);
+            const deal = await iexecPoco.viewDeal(dealId);
+            expect(deal.botFirst).equal(0);
+            expect(deal.botSize).equal(1);
+
+            // Will fail since passed taskIndex is above offset + bot size
+            await expect(iexecPocoAsAnyone.initialize(dealId, 1)).to.be.reverted;
+        });
+
+        it('Should not initialize twice', async function () {
+            const { orders } = buildOrders({
+                assets: ordersAssets,
+                requester: requester.address,
+                prices: ordersPrices,
+            });
+            const { dealId, taskId, taskIndex } = await iexecWrapper.signAndMatchOrders(orders);
+            await iexecPocoAsAnyone.initialize(dealId, taskIndex).then((tx) => tx.wait());
+            expect((await iexecPoco.viewTask(taskId)).status).equal(TaskStatusEnum.ACTIVE);
+
+            await expect(iexecPocoAsAnyone.initialize(dealId, taskIndex)).to.be.reverted;
+        });
+    });
+
+    describe('Initialize array', function () {
+        it('Should initialize array', async function () {
+            const volume = 3;
+            const { orders } = buildOrders({
+                assets: ordersAssets,
+                requester: requester.address,
+                prices: ordersPrices,
+                volume,
+            });
+            const { dealId, startTime } = await iexecWrapper.signAndMatchOrders(orders);
+            const dealIds = [dealId, dealId, dealId];
+            const taskIndexes = [0, 1, 2];
+            await time.setNextBlockTimestamp(startTime + maxDealDuration);
+
+            expect(await iexecPocoAsAnyone.callStatic.initializeArray(dealIds, taskIndexes)).to.be
```

Comment:
Does it work without `mine()` ?

Reply:
Actually we don't event need it thanks! [Remove useless time traveling](https://github.com/iExecBlockchainComputing/PoCo/pull/74/commits/4be04754df20629138bd086700c4e777eb85adb7)